manager.go

```diff
@@ -926,41 +896,15 @@ func (m *jobManager) resolveAsPullRequest(spec string) (*prowapiv1.Refs, error)
 		return nil, fmt.Errorf("when specifying a pull request, you must provide ORG/REPO#NUMBER")
 	}
 
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s/repos/%s/%s/pulls/%d", m.githubURL, url.PathEscape(locationParts[0]), url.PathEscape(locationParts[1]), num), nil)
-	if err != nil {
-		return nil, fmt.Errorf("unable to lookup pull request %s: %v", spec, err)
-	}
-	if token := os.Getenv("GITHUB_TOKEN"); len(token) > 0 {
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
-	}
-	req.Header.Add("User-Agent", "ci-chat-bot")
-	resp, err := http.DefaultClient.Do(req)
+	pr, err := m.githubClient.GetPullRequest(url.PathEscape(locationParts[0]), url.PathEscape(locationParts[1]), num)
 	if err != nil {
 		return nil, fmt.Errorf("unable to lookup pull request %s: %v", spec, err)
 	}
-	defer resp.Body.Close()
-
-	switch resp.StatusCode {
-	case 200:
-		// retrieve
-	case 403:
-		data, _ := ioutil.ReadAll(resp.Body)
-		klog.Errorf("Failed to access server:\n%s", string(data))
-		return nil, fmt.Errorf("unable to lookup pull request %s: forbidden", spec)
-	case 404:
-		return nil, fmt.Errorf("pull request %s not found", spec)
-	default:
-		return nil, fmt.Errorf("unable to lookup pull request %s: %d %s", spec, resp.StatusCode, resp.Status)
-	}
 
-	var pr GitHubPullRequest
-	if err := json.NewDecoder(resp.Body).Decode(&pr); err != nil {
-		return nil, fmt.Errorf("unable to retrieve pull request info: %v", err)
-	}
 	if pr.Merged {
 		return nil, fmt.Errorf("pull request %s has already been merged to %s", spec, pr.Base.Ref)
 	}
-	if !pr.Mergeable {
+	if !*pr.Mergable {
```

Comment:
I'm not sure if this is guaranteed to be non-nil, I'd better do if `pr.mergeable != nil && !*pr.Mergable`

Reply:
Done!